### PR TITLE
The local variable declaration is change to statement.

### PR DESCRIPTION
**Pull request by splasky**
Solve #36

### CODE REVIEW

File: tests/struct.c [hunk 1]
```diff
@@ -14,14 +14,13 @@ int main(int argc, char **argv)
 {
     struct foo bar, *p;
     struct foo *ptr;
-    char c;
-    int i;
+    char c = '1';
```
Comment (jserv) [inline tests/struct.c]: How about writing new test cases which utilize the new feature for declaration?

File: amacc.c [hunk 2]
```diff
@@ -714,13 +715,292 @@ void expr(int lev)
...
 }
 
+// Ast parsing for IR generating.
+// With a modular code generator, new targets can be easily supported such as native Arm machine code.
+// The compiler now resembles something you could actually build upon (or at least shows the way.)
+void gen(int *n)
+{
+    int i, j, k, l;
+    int *a, *b, *c, *d;
+    i = *n;
```
Comment (jserv) [inline amacc.c]: Move Line 725 into 723 for initialization.

File: amacc.c [hunk 3]
```diff
@@ -2101,4 +2084,4 @@ int main(int argc, char **argv)
     return ret;
 }
 
-// vim: set tabstop=4 shiftwidth=4 expandtab:
+// vim: set tabstop=4 shiftwidth=4 expandtab:
```
Comment (jserv) [inline amacc.c]: Don't change this line.

File: amacc.c [hunk 4]
```diff
@@ -856,137 +1136,7 @@ void stmt()
...
-    case '{':  
-        gen((int *) n[1]); gen(n + 2); break;  // parse expression or statment from ast
-    case Enter:  *++e = ENT; *++e = n[1]; gen(n + 2); if (*e != LEV) *++e = LEV; break;
-    default:
-        if (i != ';') { printf("%d: compiler error gen=%d\n", line, i); exit(-1); }
-    }
-}
-
-void die(char *msg) { printf("%s\n", msg); exit(-1); }
+void die(char *msg) { printf("codegen: %s\n", msg); exit(2); }
```
Comment (jserv) [inline amacc.c]: Why was it referred to `codegen`?

Comment (splasky) [reply amacc.c]: Oh! This is my mistake since git rebase command.

File: amacc.c (hunk 3 above)
Comment (splasky) [reply amacc.c]: Uh? This is the last comment inside amacc.c. The source line is reduce.

**Pull request merged**